v0/salesforce/transform.js

```diff
@@ -0,0 +1,191 @@
+async function responseBuilderSimple(
+  message,
+  mappingJson,
+  destination,
+  targetEndpoint
+) {
+
+  //Need to get the authorization header element
+  //Need this to be async 
+   if (!authorizationHeader) {
+     authorizationHeader = await getSFDCHeader(destination);
+   }
+ 
+  const rawPayload =  {};
+
+  //First name and last name need to be extracted from the name field
+  //and inserted into the message
+  
+  var firstName = "";
+  var lastName = "";
+  if(message.context.traits.name){
+    //Split by space and then take first and last elements
+    var nameComponents = message.context.traits.name.split(" ");
+    firstName = nameComponents[0]; //first element
+    lastName = nameComponents[nameComponents.length - 1]; //last element
+    
+  }
+
+  //Insert first and last names separately into message
+  message.context.traits.firstName = firstName;
+  message.context.traits.lastName = lastName;
+  
+  const sourceKeys = Object.keys(mappingJson);
+  sourceKeys.forEach(sourceKey => {
+    rawPayload[mappingJson[sourceKey]] = get(message, sourceKey);
+  });
+  // Remove keys with undefined values
+  const payload = removeUndefinedValues(rawPayload);
+
+  //Get custom params from destination config
+  let customParams = getParamsFromConfig(message, destination);
+  customParams = removeUndefinedValues(customParams);
+  //request configuration will be conditional
+  //POST for create, PATCH for update
+
+  const response = {
+    endpoint: targetEndpoint,
+    requestConfig: defaultPostRequestConfig,
+    header: {
+      "Content-Type" : "application/json",
+      "Authorization" : authorizationHeader
+    },
+    userId: message.anonymousId,
+    payload: { ...customParams, ...payload }
+  };
+  return response;
+};
+
+function getParamsFromConfig(message, destination) {
+  let params = {};
+
+  var obj = {};
+  // customMapping: [{from:<>, to: <>}] , structure of custom mapping
+  if (destination.Config.customMappings) {
+    destination.Config.customMappings.forEach(mapping => {
+      obj[mapping.from] = mapping.to;
+    });
+  }
+  let keys = Object.keys(obj);
+  keys.forEach(key => {
+    params[obj[key]] = get(message.properties, key);
+  });
+  return params;
+}
+
+
+
+
+//Function for handling identify events
+async function processIdentify(message,destination) {
+
+  //start with creation endpoint, update only if Lead does not exist
+  var targetEndpoint = "https://" 
+                      + destination.Config.instanceName 
+                      + ".salesforce.com"
+                      + "/services/data/v"
+                      + SF_API_VERSION
+                      + "/sobjects/Lead";
+
+  //Get the authorization header if not available
+  if (!authorizationHeader) {
+    authorizationHeader = await getSFDCHeader(destination);
+  }
+  
+  //check if the lead exists
+  //need to perform a parameterized search for this using email
+  var email = message.context.traits.email;
+  
+  var leadQueryUrl = "https://" 
```

Comment:
We should handle the case when the email is undefined. It searches for "undefined" now. Also, should we use `in=EMAIL` parameter from here to make sure we only search in email field?
https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_search_parameterized.htm

Reply:
Email is mandatory. Call should fail if eMail has not been specified. That should be in our documentation - even Segment mentions the same